components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -51,39 +53,43 @@ class Nimbus(
     appInfo: NimbusAppInfo,
     server: NimbusServerSettings?,
     errorReporter: ErrorReporter = loggingErrorReporter,
-    private val observable: Observable<NimbusInterface.Observer> = ObserverRegistry()
+    private val observable: Observable<NimbusInterface.Observer> = ObserverRegistry(),
 ) : ApplicationServicesNimbus(
     context = context,
     appInfo = appInfo,
     server = server,
     deviceInfo = NimbusDeviceInfo(
-        localeTag = Locale.getDefault().getLocaleTag()
+        localeTag = Locale.getDefault().getLocaleTag(),
     ),
     observer = Observer(observable),
     delegate = NimbusDelegate(
         dbScope = CoroutineScope(
             Executors.newSingleThreadExecutor(
-                NamedThreadFactory("NimbusDbScope")
-            ).asCoroutineDispatcher()
+                NamedThreadFactory("NimbusDbScope"),
+            ).asCoroutineDispatcher(),
         ),
         fetchScope = CoroutineScope(
             Executors.newSingleThreadExecutor(
-                NamedThreadFactory("NimbusFetchScope")
-            ).asCoroutineDispatcher()
+                NamedThreadFactory("NimbusFetchScope"),
+            ).asCoroutineDispatcher(),
         ),
         errorReporter = errorReporter,
-        logger = { logger.info(it) }
-    )
+        logger = { logger.info(it) },
+    ),
 ),
     NimbusApi,
     Observable<NimbusInterface.Observer> by observable {
     private class Observer(val observable: Observable<NimbusInterface.Observer>) : NimbusInterface.Observer {
         override fun onExperimentsFetched() {
-            observable.notifyObservers { onExperimentsFetched() }
+            MainScope().launch {
+                observable.notifyObservers { onExperimentsFetched() }
+            }
```

Comment:
I'm a bit confused about why we are using the `MainScope` directly here if we could passed via [NimbusDelegate.updateScope](https://github.com/mozilla/application-services/pull/5131/files#diff-f32b0fe10a85a5a98740efc619fc608fdc54e4b5b15a79f8afa51074cd2a3003R321)

Reply:
This is a good question. I don't think we need to do both, but I'm attracted to doing it here to make testing easier in Application Services.

Reply:
I liked James' first recommendation of using the `updateScope` because you can pass in any scope you want so a test scope can be used for runBlockingTest/testing.